pyrsistent/typing.pyi

```diff
@@ -168,21 +177,38 @@ class PList(Sequence[T], Hashable):
     def __getitem__(self, index: slice) -> PList[T]: ...
     def __hash__(self) -> int: ...
     def __len__(self) -> int: ...
-    def __lt__(self, other: PDeque[T]) -> bool: ...
+    def __lt__(self, other: PList[T]) -> bool: ...
+    def __gt__(self, other: PList[T]) -> bool: ...
     def cons(self, elem: T) -> PList[T]: ...
+    @property
+    def first(self) -> T: ...
     def mcons(self, iterable: Iterable[T]) -> PList[T]: ...
     def remove(self, elem: T) -> PList[T]: ...
+    @property
+    def rest(self) -> PList[T]: ...
     def reverse(self) -> PList[T]: ...
     def split(self, index: int) -> Tuple[PList[T], PList[T]]: ...
 
 
+# Types for Checked collections are actually determined by the __type__ dunders,
+# but this should provide better type safety than using Any
+
 class CheckedPMap(PMap[KT, VT]):
-    pass
+    def __new__(cls, source: Mapping[KT, VT] = ..., size: int = ...) -> None: ...
+    @staticmethod
+    def create(source_data: Mapping[KT_, VT_], _factory_fields: Any = ...) -> CheckedPMap[KT_, VT_]: ...
```

Comment:
Isn't `create` a class method? What's the purpose of `KT_` and `VT_` here?

Reply:
True, i missed the classmethod declaration.

I think the `staticmethod` and `classmethod` decorators require separate TypeVars, if i annotate the method as `def create(source_data: Mapping[KT, VT], _factory_fields: Any = ...) -> CheckedPMap[KT, VT]: ...`, it gives this error:

```py
class M(CheckedPMap):
    __key_type__ = int
    __value_type__ = float

a = M.create({1: 3.4})  # Dict entry 0 has incompatible type "int": "float"; expected "KT": "VT"
```

Reply:
I was thinking declaring `M` would look more like
```python
class M(CheckedPMap[int, float])):
    __key_type__ = int
    __value_type__ = float
```
in which case KT and VT work fine.

Your error message was because `M` doesn't have `KT` and `VT` filled out, so they didn't match `int` and `float`.

Using `KT_` and `VT_`, mypy says that `M.create({1: 'abc'})` returns a `CheckedPMap[int, str]`, which seems inaccurate.

We can also declare `__key_type__: Type[KT]` and `__value_type__: Type[VT]` to make sure they are the same as the type parameters.

Reply:
Please add the types of `__key_type__` and `__value_type__` I mentioned here.